app/models/user.rb

```diff
@@ -29,31 +29,30 @@ def push(tsessions, link_status, user)
 
   validates :first_name, :last_name, :email, presence: true
   validate :email_domain
+
+  after_create :add_student_role, :add_spartan_tutor_role
+
   def email_domain
     domain = email.split('@').last if email.present?
     return unless email.present? && domain != 'tamu.edu' && domain != 'spartan-tutoring.com'
 
     errors.add(:email, 'Invalid Domain. Please use your TAMU or Spartan email')
   end
 
-  def admin?
-    @role = Role.where(role_name: 'Admin')
-    (role_users.find_by role_id: @role, user_id: id) != nil
-  end
+  def add_student_role
+    domain = email.split('@').last if email.present?
+    return unless email.present? && domain == 'tamu.edu'
 
-  def tutor?
-    @role = Role.where(role_name: 'Tutor')
-    (role_users.find_by role_id: @role, user_id: id) != nil
+    @role = Role.where(role_name: 'Student')
+    roles.push(@role)
```

Comment:
Doesn't this automatically assume that _everyone_ is a student? What about tutors and admins?

Reply:
everyone that signs up with a tamu email is a student by default. if they are a spartan tutor, they don't get assigned the student role. admins can change all roles once an account is created.